### DISCUSSION TITLE
Problem w VideoDownloadHelper Companion App

### DISCUSSION THREAD

**Discussion by MichaelP-99** [aclap-dev/video-downloadhelper#463] [Help and Q&A]
### Your question

I have the VDH & Companion App license & Key and downloads worked well until the latest Firefox update was installed. However, ever since the new Firefox update, the VDH downloads YouTube vids with the QR code box watermark in the corner of footage frames. And, it's asking for the key to remove QR watermark (which I have), but the dialog is not allowing me to enter the Key in order to remove the QR watermark. How do I enter the Key to remove watermark? Is there a URL link allowing me to enter the Key?

### Addon & CoApp version and details

⊖
Video DownloadHelper 8.2.2.8 (stable)
Video DownloadHelper
Version 8.2.2.8 (stable)
Browser locale: en-US
Production build
Built on 2024-03-18
Build options: browser=firefox, linuxlic=false, noyt=false, target=mozilla
⊖
Platform
Platform Win x86-64
Browser Mozilla Firefox 125.0.3
⊖
Companion App outdated - please update
Found companion app: VdhCoApp 2.0.10
Latest available version is 2.0.19
Companion app binary: C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe

⊖
License not set
No license has been registered

*****

**Comment by jcfrog** [comment]
Hi,
This is how you can register your license key: https://github.com/aclap-dev/video-downloadhelper/wiki#register-your-premium

**Reply by MichaelP-99** [reply]
Thanks jcfrog for your fast reply. However, I have tried your suggestion (URL link) many times. It all worked fine UNTIL I reluctantly allowed Firefox to install latest update. Then everything got f'd up.  My Firefox extension profile states "ENABLED"  but the Watermark persists on all downloads. And the dialog you suggested does not work after I enter my purchased CoApp KEY. Once again, it worked great until the Firefox update was installed. I've tried several other fixes for over an hour. They don't work.

**Reply by jcfrog** [reply]
The license key is actually for the addon.
Something has probably erased the licensing info of your browser while updating.
Do you get error messages when revalidating your license key?

**Comment by MichaelP-99** [comment]
No Error Msg -- it merely keeps looping back and repeating the same dialog, eventually stating its not registered.  And the dialog box to register it won't work. Just an endless loop back around. SEE ATTACHED.

![zzz-3](https://github.com/aclap-dev/video-downloadhelper/assets/169137024/ca2c6eaa-7bb9-4018-9b22-303281500b20)
![zzz-3](https://github.com/aclap-dev/video-downloadhelper/assets/169137024/244fbb84-0c2f-4c63-9783-b071186d4fd4)
![zzz-3](https://github.com/aclap-dev/video-downloadhelper/assets/169137024/27d5198b-25b6-438c-8a53-87f2e199b1cf)

**Comment by jcfrog** [comment]
Could it be a firewall or antivirus issue?

**Comment by MichaelP-99** [comment]
jcfrog -- keep in mind that the vids do download, however they all contain that QR-code watermark in the frame corner. And the REGISTERED KEY is what removes the watermark. That part I can't seem to fix.

**Reply by jcfrog** [reply]
As long as the validation doesn't work, you are not seen as premium, so you have QR codes.

**Comment by MichaelP-99** [comment]
The same Norton 360 Virus was installed before and after the Firefox Update. My guess is something in the latest version of Firefox is ignoring the already establish KEY that always worked before.

**Reply by jcfrog** [reply]
Your license info has been erased, that's for sure. But that does not explain why you can't revalidate it. Something is blocking the validation.

**Comment by MichaelP-99** [comment]
When I paid $28 for the Key years ago, was that considered the STANDARD or thee PREMIUM version?

**Reply by jcfrog** [reply]
Premium, it means you purchased a license.

**Comment by MichaelP-99** [comment]
The puzzling part is that there IS a dialog that accommodates the re-install of the KEY. BUT that re-register dialog doesn't seem to work.

**Reply by jcfrog** [reply]
Did you try to restart your browser (closing all browser's windows)

**Comment by MichaelP-99** [comment]
Is there an easy way to back-out of this latest version of Firefox and re-instate the earlier version. Might that work?

**Comment by GrampaWildWilly** [comment]
Quoting from your own information . . .

> Found companion app: VdhCoApp 2.0.10
> Latest available version is 2.0.19

There might be a hint there.  Click the Wiki link at the top of this page & read what you find.  You'll see a reference to the latest CoApp.  That will be where you can get the latest available version, something you should get in any case.  Will it fix your registration problem?  Beats me.  I don't work for VDH.  I'm just another user.  But I've got 2.0.19 & I've registered my license just fine under Firefox 125.0.3.  Mind you, I upgraded to this version of Firefox when it came out & my license didn't unregister itself.  I'm as baffled as Jérôme is on this one.  One other possibility might be to update to the latest VDH beta.  That may or may not be totally straightforward.  I haven't checked whether a problem in the Wiki from a few weeks ago has been fixed.  But I posted full instructions on how to get the real latest VDH beta in case the one on the Wiki isn't the real one yet.  Those instructions are in this discussion: #391.

Oh.  And ask Jérôme for a refund of your second license.  The VDH license is for life.  You should never have to buy a second one.  Hmmm.......  That gives me another idea.  Which license are you trying to register, the old one or the new one?  I'm guessing you've tried both but it's worth asking about.

And . . . Being an expert at fat-fingering myself, I have to ask this:  Are you sure you're typing your license number in correctly?  You should be copy/pasting it from your registration confirmation E-mail which should make that question moot.  But it's another question worth asking.

**Reply by MichaelP-99** [reply]
ATTN:  GrampaWW & jcfrog.
An apology is in order, for my village idiot mistake. When GWW mentioned fat-finger thing -- I re-checked the ORIGINAL license receipt and compared it to my "Quick Reference" license note to myself -- and discovered it contained the wrong Key -- off by one digit. So the VDH re-enter registration DOES work when you have the correct Key. GWW's fat-finger blurb caused me to find my mistake -- THANKS!! I'm back in business again. Sorry for bothering you both by requesting your help here. Lesson learned, thanks.
